### DISCUSSION TITLE
Always get 401 Unauthorized when use the access token to access the API

### DISCUSSION THREAD

**Discussion by bugsmachine** [community/community#118908] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I am studying how to use GitHub OAuth login in my school's coursework which is a frontend and backend separated project. I am using spring boot as the backend. Now I can get the access token and always get 401 Unauthorized when using the token to apply for user information.

Here is my Java code and the error info.

![code](https://github.com/community/community/assets/107638563/fb362029-d181-4291-a965-735d620a92a4)
![error_info](https://github.com/community/community/assets/107638563/4cc04eb3-2e53-4f74-9429-fa06380f1d3e)
I tried many times, I tried "Bearer" and "bearer"... But bot not working. I do not why. I use several GitHub accounts to test and always get the same error. The token is correct and it is not expired that I am sure.

![token](https://github.com/community/community/assets/107638563/e470205d-07b0-47aa-85f7-b30eb92b115e)
How to fix this? What is my problem?

**Comment by yasindu10** [comment]
check your client ID and client secret is correct

**Reply by bugsmachine** [reply]
The client ID and client secret are correct. I can use the ID and secret to exchange the access token using the code returned by GitHub.
The error happened when I used the token to apply user data. The token seems invalid and expired because got a 401. But I CAN SURE the token is new and correct.

**Reply by yasindu10** [reply]
try adding access token before providing the content-type.

**Reply by Waklloud** [reply]
Stackoverflow

**Reply by Waklloud** [reply]
Travis

**Comment by bugsmachine** [comment, marked as answer]
I figured it out, the HTTP header for authentication is not "Bearer".  It should be "Token " and it worked after I made this modification.

**Reply by Sarahleean** [reply]
it worked error 401 gone.. but I'm now getting error 404 haha gotta go fix it as well
